packages/frameworks-sveltekit/src/lib/webauthn.ts

```diff
@@ -0,0 +1,118 @@
+import { base } from "$app/paths"
+import { startAuthentication, startRegistration } from "@simplewebauthn/browser"
+
+import type {
+  BuiltInProviderType,
+  RedirectableProviderType,
+} from "@auth/core/providers"
+import type { WebAuthnOptionsResponseBody } from "@auth/core/types"
+import type { SignInOptions, SignInAuthorizationParams } from "./client.js"
+import type { LiteralUnion } from "./types.js"
+
+/**
+ * Fetch webauthn options from server and prompt user for authentication or registration.
+ * Returns either the completed WebAuthn response or an error request.
+ *
+ * @param providerId provider ID
+ * @param options SignInOptions
+ * @returns WebAuthn response or error
+ */
+async function webAuthnOptions(providerId: string, options?: SignInOptions) {
+  const baseUrl = `${base}/auth/`
+
+  // @ts-expect-error
+  const params = new URLSearchParams(options)
+
+  const optionsResp = await fetch(
+    `${baseUrl}/webauthn-options/${providerId}?${params}`
+  )
+  if (!optionsResp.ok) {
+    return { error: optionsResp }
+  }
+  const optionsData: WebAuthnOptionsResponseBody = await optionsResp.json()
+
+  if (optionsData.action === "authenticate") {
+    const webAuthnResponse = await startAuthentication(optionsData.options)
+    return { data: webAuthnResponse, action: "authenticate" }
+  } else {
+    const webAuthnResponse = await startRegistration(optionsData.options)
+    return { data: webAuthnResponse, action: "register" }
+  }
+}
+
+/**
+ * Client-side method to initiate a webauthn signin flow
+ * or send the user to the signin page listing all possible providers.
+ * Automatically adds the CSRF token to the request.
+ *
+ * [Documentation](https://authjs.dev/reference/sveltekit/client#signin)
+ */
+export async function signIn<
```

Comment:
hmm, I can't remember why we didn't edit the `signIn` function but to create a new one when we merged for `next-auth` 🤔 do you think we should update the `signIn` function in `client` instead?

Reply:
Yeah good idea, lets do that in a follow up. We can check out the svelte + next.js use of that additional `signIn` method and potentially iterate on those in a follow-up PR